api/envoy/extensions/filters/network/connection_limit/v3/connection_limit.proto

```diff
@@ -0,0 +1,41 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.network.connection_limit.v3;
+
+import "envoy/config/core/v3/base.proto";
+
+import "google/protobuf/duration.proto";
+import "google/protobuf/wrappers.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.network.connection_limit.v3";
+option java_outer_classname = "ConnectionLimitProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Connection limit]
+// Connection limit :ref:`configuration overview <config_network_filters_connection_limit>`.
+// [#extension: envoy.filters.network.connection_limit]
+
+message ConnectionLimit {
+  // The prefix to use when emitting :ref:`statistics
+  // <config_network_filters_connection_limit_stats>`.
+  string stat_prefix = 1 [(validate.rules).string = {min_len: 1}];
+
+  // The max connections configuration to use for new incoming connections that are processed by the
+  // filter's filter chain based on per worker. When max_connection is reached, the incoming connection
```

Comment:
The use of atomics in the code below on the filter chain wide config tells me this is per-filter/filter chain and not per worker (as I would think it would be). Update?

Reply:
Updated.